controller/hydrator_dependencies.go

```diff
@@ -56,8 +57,14 @@ func (ctrl *ApplicationController) GetWriteCredentials(ctx context.Context, repo
 	return ctrl.db.GetWriteRepository(ctx, repoURL, project)
 }
 
-func (ctrl *ApplicationController) RequestAppRefresh(appName string) {
-	ctrl.requestAppRefresh(appName, CompareWithLatest.Pointer(), nil)
+func (ctrl *ApplicationController) RequestAppRefresh(appName string, appNamespace string) error {
+	// We request a refresh by setting the annotation instead of by adding it to the refresh queue, because there is no
+	// guarantee that the hydrator is running on the same controller shard as is processing the application.
+	_, err := argoutil.RefreshApp(ctrl.applicationClientset.ArgoprojV1alpha1().Applications(appNamespace), appName, appv1.RefreshTypeNormal, false)
```

Comment:
Is it expected to have "hydrate=false" here? I would naively expect that RefreshApp in the hydrator_dependencies will always want to set "hydrate=true" 🤔

Reply:
Ah sure. I'll update the comment

Reply:
Pushed a comment.

We set hydrate=true elsewhere because the idea is that we'd want to kick off a hydration any time we'd normally kick off a refresh (e.g. user hits the button or webhook indicates a new commit). It's a quick-and-dirty way to make sure we're hydrating roughly whenever we'd be doing refreshes.